lib/rspec/matchers/built_in/raise_error.rb

```diff
@@ -160,6 +160,7 @@ def warn_about_bare_error
                         "will match when Ruby raises a `NoMethodError`, `NameError` or " \
                         "`ArgumentError`, potentially allowing the expectation to pass " \
                         "without even executing the method you are intending to call. " \
+                        "Actual error raised was #{description_of(@actual_error)}. "\
```

Comment:
Yeah you're right there is no test to cover this case, I'll add one now. 

I think I can improve my work here by setting the warning at the beginning and then modifying the warning message when `@actual_error` is present.